packages/ios_platform_images/ios/Classes/UIImageIosPlatformImages.swift

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import Foundation
+import UIKit
+
+extension UIImage {
+  static func flutterImage(withName name: String) -> UIImage? {
+    let imageName = (name as NSString).lastPathComponent
+    let path = (name as NSString).deletingLastPathComponent
+    let screenScale = UIScreen.main.scale
+
+    var scaledImage: UIImage?
+
+    // Search for a scaled image in the bundle
+    // The image name may be suffixed with a scale factor, e.g. @2x
+    // If a scaled image is not found, search for standard image
+
+    for scale in stride(from: Int(screenScale), through: 1, by: -1) {
+      let scaledName = imageName.replacingOccurrences(of: ".", with: "@\(scale)x.")
```

Comment:
Why have you changed the logic and dropped the `lookupKeyForAsset` calls? This is a completely different implementation.

If you believe there's an issue with the current implementation, that needs to be resolved in a separate PR. Swift conversions PRs should not be combining conversion with logic changes, as those are completely orthogonal things.

Reply:
I test with lookupKeyForAsset in old and in new implementation and he dont works,at least it didn't work for me. I dont know why exactly, i will made some tests and try back again

Reply:
I need fix this, when i pass only image name he works.

Reply:
@stuartmorgan I reverted the implementation to how it should be and it works normally without changing the original logic. My test gave a false positive